[hotfix][table-planner] Improve error reporting when serializing CatalogTable into the plan

slinkydeveloper opened the pull request:
This is a little hotfix to improve error reporting when serializing a `CatalogTable` that cannot be serialized

twalthr reviewed (commented):
One last comment to this @slinkydeveloper.

Sorry, for being so picky here. But I think the current exception is not very useful without an object identifier or some other way of identifying the problematic table. `ContextResolvedTableJsonSerializer` should include the identifier in an exception.

slinkydeveloper commented:
@twalthr no need to be sorry. Added a try catch in the caller, which is where i have knowledge of the obj identifier, and added the class name to the original exception
